bundles/org.openhab.binding.wlanthermo/src/main/java/org/openhab/binding/wlanthermo/internal/WlanThermoNanoHandler.java

```diff
@@ -0,0 +1,245 @@
+            if (s != null)
+                updateState(channelUID, s);
+        } else {
+            if (data.setState(channelUID, command)) {
+                logger.debug("Data updated, pushing changes");
+                push();
+            } else {
+                logger.error("Could not handle command of type {} for channel {}!",command.getClass().toGenericString() , channelUID.getId());
+            }
+        }
+    }
+
+    private void update() {
+        try {
+            //Update objects with data from device
+            String json = httpClient.GET(config.getUri("/data")).getContentAsString();
+            data = gson.fromJson(json, Data.class);
+            logger.debug("Received at /data: {}", json);
+            json = httpClient.GET(config.getUri("/settings")).getContentAsString();
+            settings = gson.fromJson(json, Settings.class);
+            logger.debug("Received at /settings: {}", json);
+            
+            
+            //Update channels
+            for (Channel channel : thing.getChannels()) {
+                State state = data.getState(channel.getUID(), this);
+                if (state != null) {
+                    updateState(channel.getUID(), state);
+                } else {
+                    //if we could not obtain a state, try trigger instead
+                    String trigger = data.getTrigger(channel.getUID());
+                    if (trigger != null) {
+                        triggerChannel(channel.getUID(), trigger);
+                    }
+                }
+
+            }
+            
+
+        } catch (Exception e) {
+            logger.debug("Update failed, checking connection", e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Update failed, reconnecting...");
+            if (pollingScheduler != null) {
+                pollingScheduler.cancel(false);
+            }
+            for (Channel channel : thing.getChannels()) {
+                if (channel.getUID().getId().equals(SYSTEM + "#" + SYSTEM_ONLINE)) {
+                    updateState(channel.getUID(), OnOffType.OFF);
+                } else {
+                    updateState(channel.getUID(), UnDefType.UNDEF);
+                }
+            }
+            checkConnection();
+        }
+    }
+    
+    private void push() {
+        for (org.openhab.binding.wlanthermo.internal.api.nano.data.Channel c : data.getChannel()) {
+            try {
+                String json = gson.toJson(c);
+                logger.debug("Pushing: {}", json);
+                URI uri = config.getUri("/setchannels");
+                int status = httpClient.POST(uri)
+                                .content(new StringContentProvider(json), "application/json")
+                                .timeout(5, TimeUnit.SECONDS)
+                                .send()
+                                .getStatus();
+                if (status == 401) {
+                    logger.error(
```

Comment:
This should probably be shown to the user in the thing status. In general the user should not have to look into the log to find out what the problem is. If you set the status this log can be reduced to debug.